src/pyscipopt/scip.pyx

```diff
@@ -891,6 +891,8 @@ cdef class Model:
             return self._addLinCons(cons, **kwargs)
         elif deg <= 2:
             return self._addQuadCons(cons, **kwargs)
+        elif deg == float('inf'): # general nonlinear
+            return self._addGenNonlinearCons(cons, **kwargs)
         else:
             return self._addNonlinearCons(cons, **kwargs)
```

Comment:
so this is for polynomials only?

Reply:
`_addNonlinearCons` is for polynomials only, yes

Reply:
I would rather rename it to `_addPolyCons()` or something similar. It's quite unintuitive right now and these are only internal methods.